packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/UpdateCategoriesTests.java

```diff
@@ -0,0 +1,235 @@
+            .withCoreAPI()
+            .usingCategory(nonExistingCategory)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Try to update a non-category (folder) node and receive 400 (Bad Request)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testUpdateCategory_usingFolderNodeAndExpect400()
+    {
+        STEP("Prepare a site and a folder inside it");
+        final UserModel user = dataUser.createRandomTestUser();
+        final SiteModel site = dataSite.usingUser(user).createPublicRandomSite();
+        final FolderModel folder = dataContent.usingUser(user).usingSite(site).createFolder();
+        final RestCategoryModel categoryWithFolderId = createCategoryModelWithIdAndName(folder.getNodeRef(), getRandomName("CategoryName"));
+
+        STEP("Try to update as admin folder node as category");
+        final RestCategoryModel fixedCategoryModel = createCategoryModelOf(getRandomName("NewCategoryName"), categoryWithFolderId);
+        restClient.authenticateUser(dataUser.getAdminUser())
+            .withCoreAPI()
+            .usingCategory(categoryWithFolderId)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Try to update a root category and receive 400 (Bad Request)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testUpdateCategory_usingRootCategoryAndExpect400()
+    {
+        STEP("Create root category model");
+        final RestCategoryModel rootCategoryModel = createCategoryModelWithId(ROOT_CATEGORY_ID);
+
+        STEP("Try to update as admin root category");
+        final RestCategoryModel fixedCategoryModel = createCategoryModelOf(getRandomName("NewCategoryName"), rootCategoryModel);
+        restClient.authenticateUser(dataUser.getAdminUser())
+            .withCoreAPI()
+            .usingCategory(rootCategoryModel)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Try to update a category with an empty name and receive 400 (Bad Request)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testUpdateCategory_withEmptyNameAndExpect400()
+    {
+        STEP("Prepare as admin a category under root category");
+        final RestCategoryModel createdCategory = prepareCategoryUnderRoot();
+
+        STEP("Try to update as admin newly created category with a category without name");
+        final RestCategoryModel fixedCategoryModel = createCategoryModelOf(null, createdCategory);
+        restClient.authenticateUser(dataUser.getAdminUser())
+            .withCoreAPI()
+            .usingCategory(createdCategory)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Try to update a category using not matching ID in path and receive 400 (Bad Request)
```

Comment:
This sounds like it should be a 404.

Reply:
I will update the javadoc.

Reply:
Since we ignore other fields than `name` this test will be removed.